components/script/dom/node.rs

```diff
@@ -2244,9 +2255,11 @@ impl NodeMethods for Node {
 
         // Step 4-5.
         match node.type_id() {
-            NodeTypeId::CharacterData(CharacterDataTypeId::Text) if self.is::<Document>() => {
+            NodeTypeId::CharacterData(CharacterDataTypeId::Text(TextTypeId::Text))
```

Comment:
Probably should be `CharacterDataTypeId::Text(_)`, since spec doesn't say _exclusive_ Text node here https://dom.spec.whatwg.org/#concept-node-replace